cmake/caches/Windows-aarch64.cmake

```diff
@@ -26,21 +26,65 @@ set(LLVM_DEFAULT_TARGET_TRIPLE aarch64-unknown-windows-msvc CACHE STRING "")
 set(LLVM_ENABLE_PYTHON YES CACHE BOOL "")
-set(LLVM_RUNTIME_TARGETS
+
+set(default_targets
+      x86_64-unknown-windows-msvc
       aarch64-unknown-windows-msvc
-    CACHE STRING "")
+      i686-unknown-windows-msvc)
+set(LLVM_RUNTIME_TARGETS ${default_targets} CACHE STRING "")
+
+# Build the android builtins if NDK path is provided.
+if(NOT "$ENV{NDKPATH}" STREQUAL "")
+  list(APPEND default_targets
+       aarch64-unknown-linux-android
+       x86_64-unknown-linux-android
+       i686-unknown-linux-android
+       armv7-unknown-linux-androideabi)
+endif()
+
+set(LLVM_BUILTIN_TARGETS ${default_targets} CACHE STRING "")
+
 foreach(target ${LLVM_RUNTIME_TARGETS})
   set(RUNTIMES_${target}_LLVM_ENABLE_RUNTIMES
         compiler-rt
       CACHE STRING "")
   set(RUNTIMES_${target}_CMAKE_MT mt CACHE STRING "")
   set(RUNTIMES_${target}_CMAKE_SYSTEM_NAME Windows CACHE STRING "")
   set(RUNTIMES_${target}_CMAKE_BUILD_TYPE Release CACHE STRING "")
+  set(RUNTIMES_${target}_COMPILER_RT_BUILD_BUILTINS NO CACHE BOOL "")
   set(RUNTIMES_${target}_COMPILER_RT_BUILD_CRT NO CACHE BOOL "")
   set(RUNTIMES_${target}_COMPILER_RT_BUILD_LIBFUZZER NO CACHE BOOL "")
-  set(RUNTIMES_${target}_COMPILER_RT_BUILD_PROFILE YES CACHE BOOL "")
-  set(RUNTIMES_${target}_COMPILER_RT_BUILD_SANITIZERS NO CACHE BOOL "")
+  set(RUNTIMES_${target}_COMPILER_RT_BUILD_ORC NO CACHE BOOL "")
+  set(RUNTIMES_${target}_COMPILER_RT_BUILD_PROFILE NO CACHE BOOL "")
   set(RUNTIMES_${target}_COMPILER_RT_BUILD_XRAY NO CACHE BOOL "")
+  # Sanitizers will be configured, but not built. We have separate build
+  # steps for that, because we need a different shell for each target.
+  set(RUNTIMES_${target}_COMPILER_RT_BUILD_SANITIZERS NO CACHE BOOL "")
+endforeach()
+
+foreach(target ${LLVM_BUILTIN_TARGETS})
+  set(BUILTINS_${target}_CMAKE_MT mt CACHE STRING "")
+  if(${target} MATCHES windows-msvc)
+    set(BUILTINS_${target}_CMAKE_SYSTEM_NAME Windows CACHE STRING "")
+  elseif(${target} MATCHES linux-android)
+    # Use a single 'linux' directory and arch-based lib names on Android.
+    set(BUILTINS_${target}_LLVM_ENABLE_PER_TARGET_RUNTIME_DIR NO CACHE BOOL "")
+    set(BUILTINS_${target}_CMAKE_SYSTEM_NAME Android CACHE STRING "")
+    if(${target} MATCHES aarch64)
+      set(BUILTINS_${target}_CMAKE_ANDROID_ARCH_ABI arm64-v8a CACHE STRING "")
+    elseif(${target} MATCHES armv7)
+      set(BUILTINS_${target}_CMAKE_ANDROID_ARCH_ABI armeabi-v7a CACHE STRING "")
+    elseif(${target} MATCHES i686)
+      set(BUILTINS_${target}_CMAKE_ANDROID_ARCH_ABI x86 CACHE STRING "")
+    else()
+      set(BUILTINS_${target}_CMAKE_ANDROID_ARCH_ABI x86_64 CACHE STRING "")
+    endif()
+    set(BUILTINS_${target}_CMAKE_ANDROID_NDK $ENV{NDKPATH} CACHE PATH "")
+    set(BUILTINS_${target}_CMAKE_ANDROID_API 21 CACHE STRING "")
+    set(BUILTINS_${target}_CMAKE_C_COMPILER_TARGET "${target}21" CACHE STRING "")
+    set(BUILTINS_${target}_CMAKE_CXX_COMPILER_TARGET "${target}21" CACHE STRING "")
```

Comment:
How about re-using `${BUILTINS_${target}_CMAKE_ANDROID_API}` instead of replicating `21` here?

Reply:
Added explicit `CMAKE_ANDROID_API` with a note to keep it in sync with build.ps1 (which is unfortunate, but I don't see how to do it better)